MergeSharpWebAPI/Controllers/GraphController.cs

```diff
@@ -0,0 +1,130 @@
+using MergeSharpWebAPI.Hubs;
+using MergeSharpWebAPI.Hubs.Clients;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.SignalR;
+using Newtonsoft.Json;
+using static MergeSharpWebAPI.Globals;
+using Microsoft.AspNetCore.SignalR.Client;
+
+namespace MergeSharpWebAPI.Controllers;
+
+
+// https://stackoverflow.com/questions/12072277/reading-fromuri-and-frombody-at-the-same-time
+// https://www.strathweb.com/2013/04/asp-net-web-api-parameter-binding-part-1-understanding-binding-from-uri/
+[ApiController]
+[Route("[controller]")]
+public class GraphController : ControllerBase

```

Comment:
O I just realized this controller is gonna need a SendMessage endpoint for sending state to the frontend, when backend receives new state from server.

Reply:
Yup I'll do that in a later pr. Should have been clear about that in this pr's description